### PR TITLE
Add `Decimal.min` and `.max`

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: test/Base_Tests/src/Data/Decimal_Spec.enso [hunk 1]
```diff
@@ -663,6 +663,55 @@ add_specs suite_builder =
...
+
+            Decimal.new "0" . max (Decimal.new "-1") . should_equal (Decimal.new "0")
+            Decimal.new "0" . max (Decimal.new "1") . should_equal (Decimal.new "1")
+            Decimal.new "-1" . max (Decimal.new "0") . should_equal (Decimal.new "0")
+            Decimal.new "1" . max (Decimal.new "0") . should_equal (Decimal.new "1")
+
+            Decimal.new "12E73" . max (Decimal.new "13E60") . should_equal (Decimal.new "12E73")
+            Decimal.new "12E73" . max (Decimal.new "13E80") . should_equal (Decimal.new "13E80")
+            Decimal.new "-12E73" . max (Decimal.new "-13E60") . should_equal (Decimal.new "-13E60")
+            Decimal.new "-12E73" . max (Decimal.new "-13E80") . should_equal (Decimal.new "-12E73")
```
Comment (radeusgd) [inline test/Base_Tests/src/Data/Decimal_Spec.enso:706]: There are no tests for cases where decimals and non-decimals are intertwined.

I think both
```
Decimal.new "123" . max 100 . should_equal 123
124 . max (Decimal.new "101") . should_equal 124
```
would be good to check.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso [hunk 2]
```diff
@@ -700,6 +700,36 @@ type Decimal
...
+       Arguments:
+       - that: The number to compare `self` against.
+
+       > Example
+         Find the maximum of 12 and 13.
+
+             Decimal.new "12" . max (Decimal.new "13")
+             # => Decimal.new "13"
+    max : Number -> Number
+    max self that = if self > that then self else that
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]:
```suggestion
    max self (that : Number) = if self > that then self else that
```

It feels like we should also have the checked type ascription here - this should probably make for better widgets in the GUI (at least in future) and will aid type checking.

**Review by radeusgd — COMMENTED** [review]
I appreciate the small PRs, it makes them so much easier to review! 👍 

I'd add a few more tests for this one.

File: test/Base_Tests/src/Data/Decimal_Spec.enso (hunk 1 above)
Comment (GregoryTravis) [reply test/Base_Tests/src/Data/Decimal_Spec.enso:706]: Done

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso (hunk 2 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]: Done -- actually changed to `Decimal`; `Number` was wrong.

**Review by radeusgd — APPROVED** [review]

**Pull request merged**